### PR TITLE
Feature/apps crud

### PR DESCRIPTION
**Pull request by HubertBalcerzak**

### CODE REVIEW

File: src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+) {
+
+    @GetMapping("/apps")
+    fun applicationList(
+        principal: Principal?,
+        @PageableDefault(size = 10) pageable: Pageable
+    ): Page<ApplicationDTO> {
+        return if (principal == null) {
+            applicationService.listPublicApplications(pageable)
+                .map(applicationMapper::toApplicationDTO)
```
Comment (KocproZ) [inline src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt]: You can take the .map() outside

File: src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    @IsAdmin
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/app/{appId}")
+    fun deleteApplication(@PathVariable appId: Long) {
+        applicationService.deleteApplication(appId)
+    }
+
+    @IsAdmin
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping("/newApp")
```
Comment (KocproZ) [inline src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt]: maybe add `@RequestMapping("/app")` above the controller and make app creation on `POST /app` or `POST /app/new`

File: src/main/kotlin/me/hubertus248/deployer/applications/model/entity/ApplicationName.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+data class ApplicationName(
+    @Column(name = "name", length = 128, nullable = false, unique = true, updatable = false)
+    val value: String
+) {
+    init {
+        try {
+            require(value.isNotBlank())
+            require(value.all { it.isLetterOrDigit() || "_ -".contains(it) })
+        } catch (e: IllegalArgumentException) {
+            throw BadRequestException()
```
Comment (KocproZ) [inline src/main/kotlin/me/hubertus248/deployer/applications/model/entity/ApplicationName.kt:17]: Consider some messages to tell where the verification failed

File: src/main/kotlin/me/hubertus248/deployer/instance/controller/InstanceController.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+//todo rewrite
+@Controller
+class InstanceController(
+    private val instanceManagerService: InstanceManagerService,
+    private val instanceService: InstanceService
+) {
+
+    @IsAdmin
+    @GetMapping("/newApp")
```
Comment (KocproZ) [inline src/main/kotlin/me/hubertus248/deployer/instance/controller/InstanceController.kt:23]: same as above

**Review by KocproZ — CHANGES_REQUESTED** [review]
Look comments

File: src/main/kotlin/me/hubertus248/deployer/instance/controller/InstanceController.kt (hunk 4 above)
Comment (HubertBalcerzak) [reply src/main/kotlin/me/hubertus248/deployer/instance/controller/InstanceController.kt:23]: as stated above, this class is `//todo rewrite`

File: src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt (hunk 2 above)
Comment (HubertBalcerzak) [reply src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt]: oops, that's not very RESTful of me

File: src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt (hunk 1 above)
Comment (HubertBalcerzak) [reply src/main/kotlin/me/hubertus248/deployer/applications/controller/ApplicationController.kt]: true

File: src/main/kotlin/me/hubertus248/deployer/applications/model/entity/ApplicationName.kt (hunk 3 above)
Comment (HubertBalcerzak) [reply src/main/kotlin/me/hubertus248/deployer/applications/model/entity/ApplicationName.kt:17]: It's going to be a bit complicated to do properly, let's create an issue for now #9.

File: src/main/kotlin/me/hubertus248/deployer/instance/controller/InstanceController.kt (hunk 4 above)
Comment (KocproZ) [reply src/main/kotlin/me/hubertus248/deployer/instance/controller/InstanceController.kt:23]: "As per my last email" vibes

**Review by KocproZ — APPROVED** [review]

**Pull request merged**